bundle/edu.gemini.p2checker/src/main/java/edu/gemini/p2checker/rules/general/GeneralRule.java

```diff
@@ -347,25 +346,18 @@ private static boolean hasScienceObserves(final ConfigSequence sequence) {
         public IP2Problems check(final ObservationElements elements)  {
             if (elements.getTargetObsComp().isEmpty()) return null; // can't perform this check without a target environment
 
-            final SPTarget baseTarget = elements.getTargetObsComp().getValue().getBase();
-            if (baseTarget != null) {
-                final Target t = baseTarget.getTarget();
-                if (t instanceof SiderealTarget) {
-                    final SiderealTarget hms = (SiderealTarget) t;
-                    final scala.Option<ProperMotion> opm = hms.properMotion();
-                    if (opm.isDefined()) {
-                        final ProperMotion pm = opm.get();
-
-                        final double pm_ra = pm.deltaRA().velocity();
-                        final double pm_dec = pm.deltaDec().velocity();
-                        final double total = pm_ra * pm_ra + pm_dec * pm_dec;
-
-                        if (total > MAX_PM * MAX_PM) { //to avoid sqrt call
-                            final P2Problems problems = new P2Problems();
-                            problems.addWarning(PREFIX + "TARGET_PM_RULE", MESSAGE, elements.getTargetObsComponentNode().getValue());
-                            return problems;
-                        }
-                    }
+            final scala.Option<ProperMotion> opm = elements.getTargetObsComp().getValue().getAsterism().basePositionProperMotion();
```

Comment:
Haven't seen this definition yet, but I think perhaps the check should be applied to all targets in the asterism.  If any has a big PM then flag an error?